monix-eval/shared/src/main/scala/monix/eval/TaskLocal.scala

```diff
@@ -193,9 +194,9 @@ final class TaskLocal[A] private (default: => A) {
     *        the local gets reset to the previous value
     */
   def bindClear[R](task: Task[R]): Task[R] =
-    Task.suspend {
-      val saved = ref.value
-      ref.clear()
+    local.flatMap { r =>
```

Comment:
Does this call to `Local`? It breaks referential transparency.

Reply:
@alexandru not the `Local` object but the function in `TaskLocal`:

```scala
def local: Task[Local[A]] =
    Task.eval(ref)
```

Since we added it I thought it was better to use it when necessary.

Reply:
Sorry, I’m on my phone and the mobile GitHub interface is pretty bad.

Reply:
@alexandru no problem, it is indeed terrible.